server/lomas_server/dp_queries/dp_libraries/opendp.py

```diff
@@ -161,20 +256,47 @@ def has_dataset_input_metric(pipeline: dp.Measurement) -> None:
         raise InvalidQueryException(e)
 
 
-def reconstruct_measurement_pipeline(pipeline: str) -> dp.Measurement:
+def reconstruct_measurement_pipeline(
+    query_json: OpenDPInp, metadata: dict
+) -> dp.Measurement:
     """Reconstruct OpenDP pipeline from json representation.
 
     Args:
-        pipeline (str): The JSON string encoding of the pipeline.
+        query_json (BaseModel): The JSON request object for the query.
+        metadata (dict): The dataset metadata dictionary.\
+            Only used for polars pipelines.
 
     Raises:
-        InvalidQueryException: If the pipeline is not a measurement.
+        InvalidQueryException: If the pipeline is not a measurement or\
+            the pipeline type is not supported.
 
     Returns:
         dp.Measurement: The reconstructed pipeline.
     """
     # Reconstruct pipeline
-    opendp_pipe = make_load_json(pipeline)
+    if query_json.pipeline_type == "legacy":
+        opendp_pipe = make_load_json(query_json.opendp_json)
+    elif query_json.pipeline_type == "polars":
+        # TODO Might pickle, huge security implications!!
+        plan = pl.LazyFrame.deserialize(io.StringIO(query_json.opendp_json))
+        output_measure = {
+            "Laplace": dp.measures.max_divergence(
```

Comment:
do we want uppercase at the start ? I would have suggested "laplace" and "gaussian"

Reply:
Sure, then it's also consistent with opendp.